startup.py

```diff
@@ -0,0 +1,241 @@
+        ],
+        "linux2": [
+            # example path: /opt/hfs14.0.444/bin/houdinifx
+            "/opt/hfs{version}/bin/{executable}",
+        ]
+    }
+
+    @property
+    def minimum_supported_version(self):
+        """The minimum supported Houdini version."""
+        return "15.0"
+
+    def prepare_launch(self, exec_path, args, file_to_open=None):
+        """
+        Prepares the given software for launch
+
+        :param str exec_path: Path to DCC executable to launch
+
+        :param str args: Command line arguments as strings
+
+        :param str file_to_open: (optional) Full path name of a file to open on
+            launch
+
+        :returns: :class:`LaunchInformation` instance
+        """
+
+        # construct the path to the engine's python directory and add it to sys
+        # path. this provides us access to the bootstrap module which contains
+        # helper methods for constructing the proper environment based on the
+        # bootstrap scanario.
+        tk_houdini_python_path = os.path.join(
+            self.disk_location,
+            "python",
+        )
+        sys.path.insert(0, tk_houdini_python_path)
+
+        from tk_houdini import bootstrap
+
+        # Check the engine settings to see whether any plugins have been
+        # specified to load.
+        launch_plugins = self.get_setting("launch_builtin_plugins")
+        if launch_plugins:
+
+            # Prepare the launch environment with variables required by the
+            # plugin bootstrap.
+            self.logger.info("Launch plugins: %s" % (launch_plugins,))
+            required_env = bootstrap.get_plugin_startup_env(launch_plugins)
+
+        else:
+
+            # pull the env var names from the bootstrap module
+            engine_env = bootstrap.g_sgtk_engine_env
+            context_env = bootstrap.g_sgtk_context_env
+
+            # Prepare the launch environment with variables required by the
+            # classic bootstrap.
+            required_env = bootstrap.get_classic_startup_env()
+            required_env[engine_env] = self.engine_name
+            required_env[context_env] = sgtk.context.serialize(self.context)
+
+        # populate the file to open env. Note this env variable name existed
+        # pre software launch setup.
+        if file_to_open:
+            file_to_open_env = bootstrap.g_sgtk_file_to_open_env
+            required_env[file_to_open_env] = file_to_open
+
+        self.logger.info("Launch environment: %s" % (required_env,))
+
+        return LaunchInformation(exec_path, args, required_env)
+
+    def _scan_software(self):
+        """
+        Scan the filesystem for all houdini executables.
+
+        :return: A list of :class:`SoftwareVersion` objects.
+        """
+
+        self.logger.debug("Scanning for Houdini executables...")
+
+        # use the bundled icon
+        icon_path = os.path.join(
+            self.disk_location,
+            "resources",
+            "houdini_logo.png"
```

Comment:
can we just use the engine icon at this point instead? (i think we should do the same for the photoshop engine)